### PR TITLE
Compat with async management message receiving + Upgrade to 5.3.0-pre1

### PR DESCRIPTION
**Pull request by mathieucarbou**
Depends on https://github.com/Terracotta-OSS/terracotta-platform/pull/300

### CODE REVIEW

**Comment by mathieucarbou** [comment]
@ChrisGreenaway @anthonydahanne @ljacomet : I've released tc-platform `5.3.0-pre1`.
I've done in this PR the changes necessary for the management changes.

@ChrisGreenaway : can I let you add the necessary changes for the api change for Ehcache ? My branch is opened.

Thanks!

File: clustered/integration-test/build.gradle [hunk 1]
```diff
@@ -24,6 +24,12 @@ configurations {
 }
 
 dependencies {
+  // not required by gradle but required by the IDE because 'dist' do not has any transitive dependencies
+  testCompile project(':clustered:client')
+  testCompile project(':clustered:common')
+  testCompile project(':impl')
+  testCompile project(':xml')
+
```
Comment (chrisdennis) [inline clustered/integration-test/build.gradle]: Can we do these changes in a separate PR?

Comment (mathieucarbou) [inline clustered/integration-test/build.gradle]: yes

**Comment by anthonydahanne** [comment]
test this please
latest platform jars are now published; but I expect it will fail because of the fetchEntity change of API

File: clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java [hunk 2]
```diff
@@ -157,23 +157,19 @@ public static void beforeClass() throws Exception {
...
-    assertThat(counts.get("SERVER_ENTITY_UNFETCHED"), hasSize(3));
-
-    assertThat(readMessages(), hasSize(0));
+    waitForAllNotifications(
+      "CLIENT_CONNECTED",
+      "CLIENT_REGISTRY_AVAILABLE",
+      "CLIENT_TAGS_UPDATED",
+      "EHCACHE_CLIENT_VALIDATED",
+      "EHCACHE_RESOURCE_POOLS_CONFIGURED",
+      "EHCACHE_SERVER_STORE_CREATED", "EHCACHE_SERVER_STORE_CREATED", "EHCACHE_SERVER_STORE_CREATED",
```
Comment (chrisdennis) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java]: Would be real nice if this could be done with `nCopies(...)`.

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/ClusterTierManagerClientEntityFactory.java [hunk 3]
```diff
@@ -115,12 +115,12 @@ public void create(final String identifier, final ServerSideConfiguration config
     boolean finished = false;
 
     try {
-      EntityRef<InternalClusterTierManagerClientEntity, ClusterTierManagerConfiguration> ref = getEntityRef(identifier);
+      EntityRef<InternalClusterTierManagerClientEntity, ClusterTierManagerConfiguration, Object> ref = getEntityRef(identifier);
```
Comment (chrisdennis) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/ClusterTierManagerClientEntityFactory.java]: If we're not using this type it should probably be `Void`.

Comment (anthonydahanne) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/ClusterTierManagerClientEntityFactory.java]: ah, it could be, true

File: clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java (hunk 2 above)
Comment (mathieucarbou) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java]: I will even do it without nCopies if we could do our ehcache tests in groovy.
